Tasks/AzureRmWebAppDeployment/operations/KuduServiceUtility.ts

```diff
@@ -129,6 +132,112 @@ export class KuduServiceUtility {
+                'isAsync=true',
+                'deployer=' + VSTS_ZIP_DEPLOY
+            ];
+
+            let deploymentDetails = await this._appServiceKuduService.zipDeploy(packagePath, queryParameters);
+
+            try {
+                var kuduDeploymentDetails = await this._appServiceKuduService.getDeploymentDetails(deploymentDetails.id);
+                tl.debug(`logs from ZIP deploy: ${kuduDeploymentDetails.log_url}`);
+                await this._printZipDeployLogs(kuduDeploymentDetails.log_url);
+            }
+            catch(error) {
+                tl.debug(`Unable to fetch logs for kudu ZIP Deploy: ${JSON.stringify(error)}`)
+            }
+
+            if(deploymentDetails.status == KUDU_DEPLOYMENT_CONSTANTS.FAILED) {
+                throw tl.loc('PackageDeploymentUsingZipDeployFailed');
+            }
+
+            if(appOffline) {
+                await this._appOfflineKuduService(physicalRootPath, false);
+            }
+
+            console.log(tl.loc('PackageDeploymentSuccess'));
+            return deploymentDetails.id;
+        }
+        catch(error) {
+            tl.error(tl.loc('PackageDeploymentFailed'));
+            throw Error(error);
+        }
+    }
+
+    public async postZipDeployOperation(oldDeploymentID: string, activeDeploymentID: string): Promise<void> {
+        try {
+            tl.debug(`ZIP DEPLOY - Performing post zip-deploy operation: ${oldDeploymentID} => ${activeDeploymentID}`);
+            let manifestFile = await this._appServiceKuduService.getFileContent(`${deploymentFolder}/${oldDeploymentID}`, manifestFileName);
+            if(!!manifestFile) {
+                let tempManifestFile: string = path.join(tl.getVariable('AGENT.TEMPDIRECTORY'), manifestFileName);
+                tl.writeFile(tempManifestFile, manifestFile);
+                await this._appServiceKuduService.uploadFile(`${deploymentFolder}/${activeDeploymentID}`, manifestFileName, tempManifestFile);
+            }
+            tl.debug('ZIP DEPLOY - Performed post-zipdeploy operation.');
+        }
+        catch(error) {
+            tl.debug(`Failed to execute post zip-deploy operation: ${JSON.stringify(error)}.`);
+        }
+    }
+
+    private async _printZipDeployLogs(log_url: string): Promise<void> {
+        if(!log_url) {
+            return;
+        }
+
+        var deploymentLogs = await this._appServiceKuduService.getDeploymentLogs(log_url);
+        for(var deploymentLog of deploymentLogs) {
+            console.log(`${deploymentLog.message}`);
+            if(deploymentLog.details_url) {
+                await this._printZipDeployLogs(deploymentLog.details_url);
+            }
+        }
+    }
+
+    private async _preZipDeployOperation(): Promise<void> {
+        try {
+            tl.debug('ZIP DEPLOY - Performing pre-zipdeploy operation.');
+            let activeDeploymentID: string = await this._appServiceKuduService.getFileContent(deploymentFolder, 'active');
```

Comment:
What happens to this call when done on app on which no deployment was done earlier? will the activeDeploymentId be null? or will the call throw?

Reply:
For Newly created app, there will be no 'active' file.
The API returns 404 as status code and in such case, we return NULL. 
We then skip the pre-zip deploy operation for null active file content.